packages/dd-trace/src/exporters/agent/writer-0.4.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const encode = require('../../encode')
+const BaseWriter = require('./base-writer')
+
+class Writer extends BaseWriter {
```

Comment:
Can you clarify a bit? I'm not sure what flexibility would be gained by having one extend from the other. I think it would actually be more confusing.

Reply:
Every new version will contain changes from the previous version. As the number of changes increase, the base writer will contain more and more functionality related to all versions. Extending from the previous version allows incremental changes instead of ending up with a base class that does things specific to subsets of versions.

Reply:
That approach seems to break the "is an instance of" relationship in inheritance. A v0.5 writer is not a v0.4 writer. I think this is what's confusing me, whereas either one being an instance of some base writer makes more intuitive sense to me.

The other thing is, if I make 0.5 extend from 0.4, then either there's going to be repeated code, or there's going to be parts of 0.4 that only make sense given that we know 0.5 extends from it, or both.